pulp_ansible/tests/functional/api/collection/v3/test_sync.py

```diff
@@ -1,196 +1,139 @@
+import pytest
 
-from pulpcore.tests.functional.utils import PulpTaskError
-
-from pulp_ansible.tests.functional.utils import (
-    gen_ansible_remote,
-    monitor_task,
-)
-from pulp_ansible.tests.functional.utils import TestCaseUsingBindings
 
 REQUIREMENTS_FILE = "collections:\n  - testing.k8s_demo_collection"
 
 
-class SyncCollectionsFromPulpServerTestCase(TestCaseUsingBindings):
-    """
-    Test whether one can sync collections from a Pulp server.
+@pytest.fixture(scope="class")
+def galaxy_remote(ansible_collection_remote_factory):
+    return ansible_collection_remote_factory(
+        url="https://galaxy.ansible.com",
+        requirements_file=REQUIREMENTS_FILE,
+        sync_dependencies=False,
+    )
+
 
-    This performs two sync's, the first uses the V2 API and galaxy.ansible.com. The second is from
-    Pulp using the V3 API and uses the content brought in from the first sync.
+@pytest.fixture(scope="class")
+def galaxy_repository(ansible_bindings, monitor_task, ansible_repository_factory, galaxy_remote):
+    repository = ansible_repository_factory(remote=galaxy_remote.pulp_href)
+    result = ansible_bindings.RepositoriesAnsibleApi.sync(repository.pulp_href, {})
+    monitor_task(result.task)
+    return repository
 
+
+@pytest.fixture(scope="class")
+def galaxy_distribution(ansible_distribution_factory, galaxy_repository):
+    """
+    Provide a distribution that serves the testing.k8s_demo_collection.
     """
+    return ansible_distribution_factory(repository=galaxy_repository)
 
-    def setUp(self):
-        """Set up the Sync tests."""
-        self.requirements_file = "collections:\n  - testing.k8s_demo_collection"
-        body = gen_ansible_remote(
-            url="https://galaxy.ansible.com",
-            requirements_file=REQUIREMENTS_FILE,
-            sync_dependencies=False,
-        )
-        self.remote = self.remote_collection_api.create(body)
-        self.addCleanup(self.remote_collection_api.delete, self.remote.pulp_href)
 
-        self.first_repo = self._create_repo_and_sync_with_remote(self.remote)
-        self.distribution = self._create_distribution_from_repo(self.first_repo)
+class TestSync:
+    def test_synced_collection_is_findable(self, ansible_bindings, galaxy_distribution):
+        collection = ansible_bindings.PulpAnsibleApiV3CollectionsApi.read(
+            "k8s_demo_collection", "testing", galaxy_distribution.base_path
+        )
 
-    def test_sync_collections_from_pulp(self):
-        """Test sync collections from pulp server."""
-        second_body = gen_ansible_remote(
-            url=self.distribution.client_url,
-            requirements_file=REQUIREMENTS_FILE,
+        assert collection.name == "k8s_demo_collection"
+        assert collection.namespace == "testing"
+
+    @pytest.mark.parametrize("requirements_file", (None, REQUIREMENTS_FILE))
+    @pytest.mark.parametrize("mirror", (True, False))
+    def test_sync_collection_from_pulp(
```

Comment:
Can't this be marked parallel?

Reply:
I'll think about that later...

Reply:
Sounds good, not trivial decision to make